### PR TITLE
JAMES-2543 Uses JamesServerBuilder in MPT tests

### PR DESCRIPTION
**Pull request by jeantil**
Followup of the Junit 5 migration in #270

This commit replaces the custom xxxSmtpTestRuleFactories and all their custom server initialization logic by the JamesServerBuilder-based builders defined alongside each product. 
This way the integration tests execute against the whole product as it is assembled instead of a semi-custom assembly built especially for the tests.

### CODE REVIEW

File: mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
```
Comment (chibenwa) [inline mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml:1]: My bet is that we don't need IMAP, LMTP, POP3, etc... while testing SMTP. Maybe we can omit unneeded files?

File: mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/InMemoryDnsExtension.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    private InMemoryDNSService inMemoryDns;
+
+    @Override
+    public Module getModule() {
+        return binder -> binder.bind(DNSService.class).toInstance(inMemoryDns);
+    }
+
+    @Override
+    public void afterEach(ExtensionContext extensionContext) throws Exception {
+        inMemoryDns = null;
```
Comment (chibenwa) [inline mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/InMemoryDnsExtension.java:43]: Is this needed?

**Review by chibenwa — COMMENTED** [review]
I agree with this work.

These MPT projects had an implicit goal: demonstrate APP assembling with James, and showing we can easily write a mail server without the mailbox component. But... Undocumented, and it was merely a side effect. I fully endorse abstracting it away.

**Review by Arsnael — APPROVED** [review]
Nice work, thanks again :)

File: mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml (hunk 1 above)
Comment (jeantil) [reply mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml:1]: I tried !  If absent, the configuration mechanism will pick up the default configs in sever/protocols and try to bind on priviledged ports (such as 143), somehow despite having test-jar dependencies to the guice-cassandra and guice-cassandra-rabbit project it did not find the files there. I gave up after wasting a full day trying to trace it :(

File: mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/InMemoryDnsExtension.java (hunk 2 above)
Comment (jeantil) [reply mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/InMemoryDnsExtension.java]: nope

File: mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml (hunk 1 above)
Comment (chibenwa) [reply mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml:1]: Ok. I though the default was disabled.

Maybe instead we can pass an empty configuration block?

Comment (jeantil) [reply mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/resources/imapserver.xml:1]: >Ok. I though the default was disabled.
It definitely isn't for IMAP I didn't try for the others

>Maybe instead we can pass an empty configuration block?
The idea never crossed my ind but it does work rather well and is much more satisfying ! fixedup

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Hi @jeantil 

This has been merged. Thanks again a lot for this!

Can you help me close this PR? Thank you!

**Pull request closed without merging**